app/services/txt2sql_engine.py

```diff
@@ -324,4 +324,22 @@ def txt2sql_engine(self,user,user_prompt,thread_id):
         else:
             response={"result":explanation}
             result={'success':False,'message':explanation,'data':[response]} 
-        return result
\ No newline at end of file
+        return result
+    
+    def get_table_columns(self, user, table_name):
+        """
+        Returns the list of columns for a given table in the user's database schema.
+        """
+        db_schema_json = self.app_db.get_user_db(user[8])
+        if not db_schema_json or len(db_schema_json) < 3:
+            return []
+        schema = db_schema_json[2]
```

Comment:
```pythondb_schema_json = self.app_db.get_user_db(user[8])
if not db_schema_json or len(db_schema_json) < 3:
    return []
schema = db_schema_json[2]```` ``criticality: MAJOR `` 

 > Using magic number indexing (user[8], db_schema_json[2]) makes code fragile and hard to maintain. Consider using named constants or dictionary keys for better readability and maintainability.